packages/clerk-js/src/core/tokenCache.ts

```diff
@@ -102,9 +99,10 @@ const MemoryTokenCache = (prefix = KEY_PREFIX): TokenCache => {
     }
 
     const nowSeconds = Math.floor(Date.now() / 1000);
-    const willExpireSoon = value.expiresAt && value.expiresAt - leeway < nowSeconds;
```

Comment:
@nikosdouvlis I feel that expiresAt and expiresIn variables were easier to read and follow in the algorithm. They also map better to the JWT lingo that drives the algorithm. 

What was the bug with the original algorithm?

Reply:
We used to compare the `expiresAt` that was generated on GCP with the user's own clock. That led to the clock skew issue that a user reported on intercom. Their clock was about 1minute behind GCP, so the `expiresAt - nowSeconds;` was always more than the token TTL.

According to your comment, we can define expiresAt as such:
```
value.expiresAt = nowSeconds  + ttl; 
```
We calculate the TTL and then we add it to the time the request for the new token was made. Does this make more sense to you?

Reply:
@SokratisVidros PTAL (review it per commit, it will make more sense)